### PR TITLE
Add Parsing Support for Extensions.cfg

### PR DESCRIPTION
**Pull request by cwfitzgerald**
Pretty standard PR at this point. Adds extensions.cfg support, expands the kvp derives, adds it to the corpus tester, fixes the bugs found in the wild.

### CODE REVIEW

File: bve-corpus/src/bin/bve-parser-run.rs [hunk 1]
```diff
@@ -13,7 +14,8 @@ arg_enum! {
         B3D,
         CSV,
         Animated,
-        TrainDat
+        TrainDat,
+        ExtensionsCfg,
```
Comment (DethRaid) [inline bve-corpus/src/bin/bve-parser-run.rs]: yay trailing comma so this line won't be marked as changed next pr

Comment (cwfitzgerald) [reply bve-corpus/src/bin/bve-parser-run.rs]: I blame the fact it's in a macro so rustfmt doesn't touch it :)

File: bve-derive/src/kvp.rs [hunk 2]
```diff
@@ -60,6 +73,18 @@ fn parse_fields(item: &ItemStruct) -> Vec<Field> {
                         _ => unimplemented!(),
                     }
                 }
+                if last_path_str == "HashMap" {
+                    field.kind = FieldKind::Hash;
+                    match &segments.arguments {
+                        PathArguments::AngleBracketed(angled) => {
+                            match angled.args.last().expect("Generic must have args") {
+                                GenericArgument::Type(ty) => field.ty = ty.clone(),
+                                _ => unimplemented!(),
```
Comment (DethRaid) [inline bve-derive/src/kvp.rs]: You're only handling one kind of argument here? Do you have plans to handle others?

Comment (cwfitzgerald) [reply bve-derive/src/kvp.rs]: Nope. If I need to, I will, but I doubt it. This macro only should implement exactly what I need, no more.

**Pull request merged**